modules/virtual_network_gateway/variables.tf

```diff
@@ -293,36 +338,69 @@ variable "local_network_gateways" {
       ]
     }
   }
-
+  ```
   EOF
-  type        = any
+  type = map(object({
+    local_ng_name   = string
+    connection_name = string
+    remote_bgp_settings = optional(list(object({
+      asn                 = string
+      bgp_peering_address = string
+      peer_weight         = optional(number)
+    })))
+    gateway_address = optional(string)
+    address_space   = optional(list(string))
+    custom_bgp_addresses = optional(list(object({
+      primary   = string
+      secondary = optional(string)
+    })))
+  }))
 }
 
-variable "ipsec_shared_key" {
-  description = "The shared IPSec key."
+# Virtual Network Gateway connection
+variable "connection_type" {
+  description = "The type of VNG connection."
+  default     = "IPsec"
   type        = string
-  sensitive   = true
+  validation {
+    condition     = contains(["IPsec", "ExpressRoute", "Vnet2Vnet"], var.connection_type)
+    error_message = "Valid options are IPsec (Site-to-Site), ExpressRoute (ExpressRoute), and Vnet2Vnet (VNet-to-VNet)"
+  }
 }
 
 variable "connection_mode" {
-  description = "Connection mode to use. Possible values are Default, InitiatorOnly and ResponderOnly. Defaults to Default. Changing this value will force a resource to be created."
+  description = "Connection mode to use."
+  default     = "Default"
   type        = string
+  validation {
+    condition     = contains(["Default", "InitiatorOnly", "ResponderOnly"], var.connection_mode)
```

Comment:
this validation could fail if you assign `null`

Reply:
try adding `nullable=false`

Reply:
added, thanks